x-pack/plugins/security_solution/public/management/cypress/e2e/endpoint/artifacts.cy.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { recurse } from 'cypress-recurse';
+import { getArtifactsListTestsData } from '../../fixtures/artifacts_page';
+import { API_HEADER, removeAllArtifacts } from '../../tasks/artifacts';
+import { loadEndpointDataForEventFiltersIfNeeded } from '../../tasks/load_endpoint_data';
+import { login } from '../../tasks/login';
+import { performUserActions } from '../../tasks/perform_user_actions';
+
+const yieldEndpointPolicyRevision = (): Cypress.Chainable<number> =>
+  cy
+    .request({
+      method: 'GET',
+      url: '/api/fleet/package_policies?kuery=ingest-package-policies.package.name%3A%20endpoint',
+      headers: API_HEADER,
+    })
+    .then(({ body }) => {
+      return body.items?.[0]?.revision ?? -1;
+    });
+
+const yieldInstalledEndpointRevision = (): Cypress.Chainable<number> =>
+  cy
+    .request({ method: 'GET', url: '/api/endpoint/metadata', headers: API_HEADER })
+    .then(({ body }) => {
+      expect(body.data.length).is.lte(1); // during update it can be temporary zero
+      return body.data?.[0]?.policy_info?.endpoint?.revision ?? -1;
+    });
+
+const parseRevNumber = (revString: string) => Number(revString.match(/\d+/)?.[0]);
+
+describe('Artifact pages', () => {
+  before(() => {
+    login();
+    loadEndpointDataForEventFiltersIfNeeded();
+    removeAllArtifacts();
+
+    // wait for ManifestManager to pick up artifact changes that happened either here
+    // or in a previous test suite `after`
+    cy.wait(6000); //  packagerTaskInterval + 1s
+
+    yieldEndpointPolicyRevision().then((actualEndpointPolicyRevision) => {
+      const hasReachedActualRevision = (revision: number) =>
+        revision === actualEndpointPolicyRevision;
+
+      // need to wait until revision is bumped to ensure test success
+      recurse(yieldInstalledEndpointRevision, hasReachedActualRevision, { delay: 1500 });
+    });
+  });
```

Comment:
We need to stabilise the Endpoint policy revision number.

See the following 2 cases:
- if a previously running test creates artifacts and leaves them, that's easy: we can check if we need to remove any artifact, and if yes, then delete and wait for a revision bump (the previous version did this: 59ab081449bc1a091b354d5a74d3f96168338206)
- but, if the previously running test clears the artifacts in its `after` hook, we don't know if we need to wait or not (at least I couldn't find a way), so the test simply waits 6 seconds to make sure that `ManifestManager` (which runs every 5 seconds) picks up the artifact change, and then waits until the installed revision equals to the revision in the policy

Is there maybe a better solution? @paul-tavares @patrykkopycinski

Reply:
In my opinion, we could move the code responsible for checking that the endpoint has the correct revision number to `after()` hook, so we are sure the next test can run without the need to add any extra checks in `before()`.
I've checked that and I think the best way to determine if there are any pending configs we should not only verify endpoint policy revision but also compare agent policy `configured` vs `applied` that we can retrieve from the same endpoint.
However, during my investigation, I've found out that there is a bug that always returns `0` for the `applied.policy_version`
https://github.com/elastic/kibana/compare/main...patrykkopycinski:test/e2e-endpoint-tests-for-policy-updates2?expand=1#diff-edce5ff8c9983cc8cf7e591f677401a9174d64a4398a6eb3f70b5ae021ccedeeR434

Reply:
@patrykkopycinski thank you for the input! Regarding moving the check to `after()`: I think that would mean that we need to add that check to every other test groups in this suite, because we cannot be sure which test group runs before this one. Instead of this, I think using the `before()` hook is usually safer, because this way we are independent of other test groups.

Regarding `configured` vs `applied`: you are right, thanks! I'll update it.

Regarding the `0` bug: it looks okay for me. I'll check it again after merging the latest main.

Reply:
@patrykkopycinski, I didn't want to incorporate the fix for the `0` policy version in this test-only PR, so here's a separate one: #152369